pyiceberg/table/__init__.py

```diff
@@ -1578,6 +1595,120 @@ def to_ray(self) -> ray.data.dataset.Dataset:
         return ray.data.from_arrow(self.to_arrow())
 
 
+class IncrementalAppendScan(DataScan):
```

Comment:
I'm a bit confused here. Should I modify the inheritance structure? The current inheritance structure is TableScan -> DataScan -> IncrementalAppendScan. Should I create a class, like BaseScan, and make both DataScan and IncrementalAppendScan inherit from it,  and move the `snapshot_id` `snapshot()` from TableScan into DataScan? Like:
```
TableScan  
    >BaseScan 
         >DataScan 
         >IncrementalAppendScan
```